base/sparse/cholmod.jl

```diff
@@ -1,143 +1,126 @@
 
-importall Base.SparseMatrix
-import Base.SparseMatrix: increment, increment!, decrement, decrement!
+#########
+# Setup #
+#########
 
 include("cholmod_h.jl")
 
-macro isok(A)
-    :($A==CHOLMOD_TRUE || throw(CholmodException()))
-end
+## macro to generate the name of the C function according to the integer type
+macro cholmod_name(nm,typ) string("cholmod_", eval(typ) == Int64 ? "l_" : "", nm) end
 
-const chm_ver    = Array(Cint, 3)
-if dlsym(dlopen("libcholmod"), :cholmod_version) != C_NULL
-    ccall((:cholmod_version, :libcholmod), Cint, (Ptr{Cint},), chm_ver)
-else
-    ccall((:jl_cholmod_version, :libsuitesparse_wrapper), Cint, (Ptr{Cint},), chm_ver)
-end
-const chm_com_sz = ccall((:jl_cholmod_common_size,:libsuitesparse_wrapper),Int,())
-const chm_com    = fill(0xff, chm_com_sz)
-const chm_l_com  = fill(0xff, chm_com_sz)
-## chm_com and chm_l_com must be initialized at runtime because they contain pointers
-## to functions in libc.so, whose addresses can change
-function cmn(::Type{Int32})
-    if isnan(reinterpret(Float64,chm_com[1:8])[1])
-        @isok ccall((:cholmod_start, :libcholmod), Cint, (Ptr{UInt8},), chm_com)
-    end
-    chm_com
-end
-function cmn(::Type{Int64})
-    if isnan(reinterpret(Float64,chm_l_com[1:8])[1])
-        @isok ccall((:cholmod_l_start, :libcholmod), Cint, (Ptr{UInt8},), chm_l_com)
+for Ti in IndexTypes
+    @eval begin
+        function common(::Type{$Ti})
+            a = fill(0xff, cholmod_com_sz)
+            @isok ccall((@cholmod_name "start" $Ti
+                , :libcholmod), Cint, (Ptr{UInt8},), a)
+            set_print_level(a, 0) # no printing from CHOLMOD by default
+            a
+        end
     end
-    chm_l_com
 end
 
-# check the size of SuiteSparse_long
-if int(ccall((:jl_cholmod_sizeof_long,:libsuitesparse_wrapper),Csize_t,())) == 4
-    const CholmodIndexTypes = (:Int32, )
-    typealias CHMITypes Union(Int32)
-else
-    const CholmodIndexTypes = (:Int32, :Int64)
-    typealias CHMITypes Union(Int32, Int64)
+### These offsets are defined in SuiteSparse_wrapper.c
+const cholmod_com_offsets = Array(Csize_t, 19)
+ccall((:jl_cholmod_common_offsets, :libsuitesparse_wrapper),
+      Void, (Ptr{Csize_t},), cholmod_com_offsets)
+const common_final_ll = (1:4) + cholmod_com_offsets[7]
+const common_print = (1:4) + cholmod_com_offsets[13]
+const common_itype = (1:4) + cholmod_com_offsets[18]
+const common_dtype = (1:4) + cholmod_com_offsets[19]
+
+function set_print_level(cm::Array{UInt8}, lev::Integer)
+    cm[common_print] = reinterpret(UInt8, [int32(lev)])
 end
 
+####################
+# Type definitions #
+####################
 
-typealias CHMVTypes Union(Complex128, Float64)
-typealias CHMVRealTypes Union(Float64)
+# The three core data types for CHOLMOD: Dense, Sparse and Factor. CHOLMOD manages the memory, so the Julia versions only wrap a pointer to a struct. Therefore finalizers should be registret each time a pointer is returned from CHOLMOD.
```

Comment:
Can you put in some newlines for this and other really long comment lines?

And I'm not so sure whether that 64 bit appveyor failure is spurious, I'm building this locally and will let you know.